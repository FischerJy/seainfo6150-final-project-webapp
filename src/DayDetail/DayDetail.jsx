import React, { createRef, useEffect, useState } from "react";
import dayjs from "dayjs";
import { useParams } from "react-router";
import { Link } from "react-router-dom";
import Card from "../Card/Card";

export default function DayDetail() {
  const { lat, lon, name, dt } = useParams();

  const [info, setInfo] = useState({
    wind: {},
    coord: {},
    main: {},
    weather: [{}],
  });

  useEffect(() => {
    var getlist = ({ lat, lon, dt }) => {
      fetch(
        `https://api.openweathermap.org/data/2.5/onecall?dt=${dt}&lat=${lat}&lon=${lon}&exclude=current,minutely,hourly,alerts&appid=d2c0715f020c3a6cd3e893289a84cbc7&units=metric`
      )
        .then((res) => res.json())
        .then((res) => {
          console.log("res -> :", res);
          setInfo(res.current);
        });
    };
    getlist({ lat, lon, dt });
    return () => {
      getlist = () => {};
    };
  }, [lat, lon, dt]);

  return (
    <div className="main">
      <br />
      <br />
      <div>
        <h1>
          {name} Weathers Of {dayjs(dt * 1000).format("YYYY-MM-DD")}
        </h1>

        <div className="card-wrap-a" >
          <div>
            <p>temp : {info.temp}</p>
            <p>wind_speed : {info.wind_speed}</p>
            <p>clouds : {info.clouds}%</p>
            <p>feels_like : {info.feels_like}</p>
            <p>visibility : {info.visibility}m</p>
          </div>
        </div>
      </div>
    </div>
  );
}
