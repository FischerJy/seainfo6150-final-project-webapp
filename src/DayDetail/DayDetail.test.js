import React from "react";
import { render } from "@testing-library/react";
import DayDetail from "./DayDetail.jsx";
import { BrowserRouter } from "react-router-dom";



it("renders the DayDetail component correctly", () => {
    const { container } = render(<BrowserRouter>
        <DayDetail />
    </BrowserRouter>);
    expect(container).toMatchSnapshot();
});