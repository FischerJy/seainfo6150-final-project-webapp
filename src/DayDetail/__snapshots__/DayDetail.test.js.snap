// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the DayDetail component correctly 1`] = `
<div>
  <div
    class="main"
  >
    <br />
    <br />
    <div>
      <h1>
         Weathers Of 
        Invalid Date
      </h1>
      <div
        class="card-wrap-a"
      >
        <div>
          <p>
            temp : 
          </p>
          <p>
            wind_speed : 
          </p>
          <p>
            clouds : 
            %
          </p>
          <p>
            feels_like : 
          </p>
          <p>
            visibility : 
            m
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
